java-manta-client/src/main/java/com/joyent/manta/client/MantaSeekableByteChannel.java

```diff
@@ -284,15 +274,16 @@ public long size() throws IOException {
             throw new ClosedChannelException();
         }
 
-        final HttpResponse response = connectOrGetResponse();
-        final HttpEntity entity = response.getEntity();
-        final long contentLength = entity.getContentLength();
+        final MantaObjectInputStream stream = connectOrGetResponse();
+        final long contentLength = stream.getContentLength();
 
         if (contentLength == UNKNOWN_CONTENT_LENGTH) {
             MantaClientException e = new MantaClientException(
                     "Can't get SeekableByteChannel for objects of unknown size");
             HttpUriRequest request = requestRef.get();
             if (request != null) {
+                @SuppressWarnings("unchecked")
+                HttpResponse response = (HttpResponse)stream.getHttpResponse();
```

Comment:
This is returned as an untyped object because when building the JAR, we shade Apache HTTP Client. This will change the object signature to com.joyent.org.apache.http.*

In order to get this value to be passed between packages, it needs to be public, but we don't want to expose the underlying API for the library because it may change out from under people's feet in the future. I've seen this pattern before. Untyped objects are occasionally available publicly for debugging if an implementor wants it, but there are no guarantees as to what object will actually be there.